product_pack/models/product_template.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    pack_type = fields.Selection([
+        ('components_price', 'Detailed - Components Prices'),
+        ('totaliced_price', 'Detailed - Totaliced Price'),
+        ('fixed_price', 'Detailed - Fixed Price'),
+        ('none_detailed_assisted_price', 'None Detailed - Assisted Price'),
+        ('none_detailed_totaliced_price', 'None Detailed - Totaliced Price'),
+    ],
+        'Pack Type',
+        help="On sale orders or purchase orders:\n"
+             "* Detailed - Components Prices: Detail lines with prices.\n"
+             "* Detailed - Totaliced Price: Detail lines totalicing "
+             "lines prices on pack (don't show component prices).\n"
+             "* Detailed - Fixed Price: Detail lines and use product"
+             " pack price (ignore line prices).\n"
+             "* None Detailed - Assisted Price: Do not detail lines. "
+             "Assist to get pack price using pack lines./n"
+             "* None Detailed - Totaliced Price: Do not detail lines. "
+             "Do not assist to get pack price using pack lines."
+    )
+    pack = fields.Boolean(
```

Comment:
Do we really need this field? :thinking: 

You can consider a product without `pack_type` as a non-pack product, and then get rid of this field. I also think it would be even more easy to use.

Reply:
From UX point of view, it's better to have this flag, as it's a switch for the rest. I see correct to keep it, although I understand your reasoning. Another UX option would be to have an option `No pack` as default and make `pack_type` as required, but then that advanced selection will be outside the page and out of context.